### PR TITLE
add cpuArchitecture filter to nodeResources collector

### LINKED ISSUE #805: Add a new check to verify if the platform  is supported and node affinity criteria
**Describe the rationale for the suggested feature.**

See that kots only ships amd64 images so far https://github.com/replicatedhq/kots/issues/3360. 
Also, it seems to have some nodeAffinity to verify the labels as well: https://github.com/replicatedhq/kots/issues/3361

**Describe the feature**

Therefore, the idea here would to be proposed a new check to verify the cluster node platform and the labels (nodeAffinity criteria)

### PR DESCRIPTION
**Pull request by adamancini**
## Description, Motivation and Context

support filtering nodes based on CPU architecture in `nodeResources` analyzer

resolves #805 

## Checklist

- [x] New and existing tests pass locally with introduced changes.
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] The commit message(s) are informative and highlight any breaking changes
- [x] Any documentation required has been added/updated. For changes to https://troubleshoot.sh/ create a PR [here](https://github.com/replicatedhq/troubleshoot.sh/pulls)

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: pkg/analyze/node_resources.go [hunk 1]
```diff
@@ -351,6 +351,14 @@ func nodeMatchesFilters(node corev1.Node, filters *troubleshootv1beta2.NodeResou
 		}
 	}
 
+	if filters.CPUArchitecture != "" {
```
Comment (banjoh) [inline pkg/analyze/node_resources.go]: Please add some unit tests

**Review by banjoh — COMMENTED** [review]
Changes look good to me

**Comment by banjoh** [comment]
Run `make generate`

File: pkg/analyze/node_resources.go (hunk 1 above)
Comment (adamancini) [reply pkg/analyze/node_resources.go]: done

**Comment by adamancini** [comment]
> Run `make generate`

done

**Review by xavpaice — DISMISSED** [review]
LGTM but needs `make schemas` in order to pass tests

**Review by banjoh — APPROVED** [review]
LGTM. Ship it :)

**Comment by banjoh** [comment]
> > Run `make generate`
> 
> done

I meant `make schemas` which you have done ;)

**Pull request merged**